core/src/main/java/org/apache/iceberg/jdbc/JdbcCatalog.java

```diff
@@ -250,10 +266,111 @@ public void setConf(Configuration conf) {
     this.conf = conf;
   }
 
+  private boolean insertProperties(Namespace namespace, Map<String, String> properties) {
+    String namespaceName = JdbcUtil.namespaceToString(namespace);
+
+    try {
+      int insertedRecords = connections.run(conn -> {
+        StringBuilder sqlStatement = new StringBuilder(JdbcUtil.INSERT_NAMESPACE_PROPERTIES_SQL);
+
+        for (int i = 0; i < properties.size() - 1; i++) {
```

Comment:
Nit: This took me quite some time to read and realize what was taking place here. Possibly it could be cleaned up a bit to help readers?

The `INSERT_NAMESPACE_PROPERTIES_SQL = `INSERT INTO iceberg_namespace_properties(catalog_name , namespace, key, value) VALUES (?,?,?,?)`.


And then the code iterates over `properties.size() - 1`, appending `, (?,?,?,?)`.

So a few concerns:
1) What happens if this function is called when `properties` is empty? Is that possible?
2) For readability purposes, would it make sense to possibly append first the INSERT INTO statement, then use a google comon's library JOINER of `,` or a repeater to join properties.size() count of placeholders?

I had to spend quite some time verifying the correctness of this and I'm still not entirely sure (though I assume it's not going to be called with an empty `properties` value.

Reply:
It took me a bit of work to get the String parsing and the indexing right as well. I can clean it up with some of your JOINER suggestions.
The checks are present to prevent this from being called with null or empty properties.

Reply:
Yeah completely understand that formatting Strings like this is not as straightforward in vanilla Java.

Check out the joiner. One simpler approach might be pushing a collection of Strings of `(?,?,?,?)` for each property pair and then using the `org.apache.iceberg.relocated.com.google.common.base.Joiner` on that collection to get the final 

It's definitely not as convenient as Scala's `.mkString(",")` etc.

Reply:
I see that. But I doubt there's much advantage to adding a separate string collection and then joining to the main string than doing it with string builder directly.

Reply:
Arguably there's not in terms of performance. That was just one quick way I thought of doing it. There are other things in the relocated guava lib that might help, but I'm not an expert in those by any means either.

It's more of a readability concern. What was confusing for me was that the `INSERT_NAMEPACES_PROPERTIES_SQL` has the contents of `INSERT_PROPERTIES_VALUES_BASE` appended at the end. And then we continue to append it.

Maybe just remove the `INSERT_PROPERTIES_VALUES_BASE` from the `INSERT_NAMESPACES_PROPERTIES_SQL`, and instead explicitly append `INSERT_PROPERTIES_VALUES_BASE` at the beginning of the loop, then keep the rest of the loop (with further commas and appends) as is?

For me, it would be much more readable that way, but not add the additional performance overhead.

The for loop being one less than expected pops out, so making the explicit additional call would, imho, make it more clear that this is intentional in the for loop and why it's intentional.

This path isn't really hot by any means, so one extra call to string builder's append isn't going to cause any real performance degradation that's not already there.

Just my 2 cents.

Reply:
This makes sense. I like the idea of fixing the loop to make it all values in one go.